crates/oxc_linter/src/rules/eslint/no_global_assign.rs

```diff
@@ -0,0 +1,111 @@
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+
+use crate::{context::LintContext, globals::BUILTINS, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-global-assign): Read-only global '{0}' should not be modified.")]
+#[diagnostic(severity(warning))]
+struct NoGlobalAssignDiagnostic(
+    Atom,
+    #[label("Read-only global '{0}' should not be modified.")] pub Span,
+);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoGlobalAssign {
+    excludes: Vec<Atom>,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow modifications to read-only global variables.
+    ///
+    /// ### Why is this bad?
+    /// In almost all cases, you don’t want to assign a value to these global variables as doing so could result in losing access to important functionality.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// Object = null
+    /// ```
+    NoGlobalAssign,
+    correctness
+);
+
+impl Rule for NoGlobalAssign {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let obj = value.get(0);
+
+        Self {
+            excludes: obj
+                .and_then(|v| v.get("exceptions"))
+                .and_then(serde_json::Value::as_array)
+                .unwrap_or(&vec![])
+                .iter()
+                .map(serde_json::Value::as_str)
+                .filter(std::option::Option::is_some)
+                .map(|x| Atom::from(x.unwrap().to_string()))
+                .collect::<Vec<Atom>>(),
+        }
+    }
+
+    fn run<'a>(&self, _node: &oxc_semantic::AstNode<'a>, ctx: &LintContext<'a>) {
```

Comment:
This is run on every ast node, you can use the run_on_symbol function.

Reply:
OK, I feel that there are two problems here.
1. It seems that global variables that is not resolved can not be executed by "run_on_symbol". 🤔 I may need some help in this place
2. global variables should be divided into different environments such as browser or node as eslint test cases shown. [eslint-test-case](https://github.com/eslint/eslint/blob/853d32baa8934c08b59a738470b72522e1505f6f/tests/lib/rules/no-global-assign.js#L27C8-L27C57)

Reply:
For 1, let me add a `run_once` function.
For 2, we can revisit this when we have a "Environments" option.

Reply:
@SoonIter Checkout https://github.com/web-infra-dev/oxc/pull/647/files